### PR TITLE
Add magazines to weekly debrief

### PR DESCRIPTION
**Pull request by isaachan100**
## Overview
Added magazines to weekly debrief feature
Add jest testing for magazines and users

## Changes Made
- updated weekly debrief and user model types
- changed logic in controllers to support magazines in weekly debrief
- created mutation: readMagazine(...): User
- updated mutation: getWeeklyDebrief(...): [WeeklyDebrief!]!

Miscellaneous
- added rules to eslintrc to suppress `import/named` and `import/prefer-default-export` warnings
- updated gitignore to include secrets folder

Jest testing:
- added `MagazineRepo` test cases for `getAllMagazines`, `getMagazine(s)ByID(s)`, `getMagazinesByPublicationSlug(s)`, `getMagazinesBySemester`, `getFeaturedMagazines`, and `incrementShoutouts`
- added `UserRepo` test cases for `getUserByUUID`, `(un)followPublication`, `incrementShoutouts`, `incrementBookmarks`, `appendReadMagazine`, and `appendReadArticle`.
- tested Weekly Debrief indirectly through UserRepo test cases

## Test Coverage

ran the two new mutations locally and verified the outputs
all jest test cases passed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +50,9 @@
     "rss-parser": "^3.9.0",
     "ts-jest": "^29.0.3",
     "type-graphql": "^1.1.1",
+    "typescript": "^4.0.5",
```
Comment (snajima) [inline package.json:53]: why are we downgrading? document a reason in your pr description or revert this change

File: package.json [hunk 2]
```diff
@@ -60,10 +63,10 @@
     "@types/node": "^14.11.4",
     "eslint": "^7.12.1",
     "eslint-config-airbnb-typescript-prettier": "^3.1.0",
+    "eslint-plugin-import": "^2.27.5",
```
Comment (snajima) [inline package.json:66]: what's all the eslint stuff for? also needs docoumenting

**Review by snajima — CHANGES_REQUESTED** [review]
great second PR! let's create test cases for the Magazine/new WeeklyDebrief model and address some of the mentioned issues to get this merged!

**Review by SashaLoayza — CHANGES_REQUESTED** [review]
I agree with Shungo that we shouldn't downgrade the typescript version. If adding the line to ignore import/named error in .eslintrc works, we should do that.

**Comment by katesliang** [comment]
prod

**Review by snajima — APPROVED** [review]
good job isaac 🎉 !! super proud

**Review by SashaLoayza — APPROVED** [review]

**Pull request merged**